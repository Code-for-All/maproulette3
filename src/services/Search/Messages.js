import { defineMessages } from 'react-intl'

/**
 * Internationalized messages for use with Sort.
 */
export default defineMessages({
  name: {
    id: "Challenge.sort.name",
    defaultMessage: "Name",
  },
  created: {
    id: "Challenge.sort.created",
    defaultMessage: "Newest",
  },
  popularity: {
    id: "Challenge.sort.popularity",
    defaultMessage: "Popular",
  },
  default: {
    id: "Challenge.sort.default",
    defaultMessage: "Default",
  },
})
