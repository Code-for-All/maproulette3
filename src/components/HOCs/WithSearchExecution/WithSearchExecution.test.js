import React, { Component } from 'react'
import _isFunction from 'lodash/isFunction'
import { _WithSearchExecution, mapDispatchToProps }
       from './WithSearchExecution'
import { performSearch } from '../../../services/Search/Search'

jest.mock('../../../services/Search/Search')

let basicState = null
let WrappedComponent = null

beforeEach(() => {
  basicState = {
    searchQueries: {
      searchName: "my search"
    }
  }

  WrappedComponent = _WithSearchExecution(
    () => <div className="child" />,
    "searchName",
    () => "my function"
  )
})

test("searchQueries.searchName is passed through to the wrapped component", () => {
  const wrapper = shallow(
    <WrappedComponent {...basicState} />
  )

  expect(
    wrapper.props().searchQueries.searchName
  ).toBe(basicState.searchQueries.searchName)

  expect(wrapper).toMatchSnapshot()
})

test("fetchResults function is passed through to the wrapped component", () => {
  const myPerformSearch = jest.fn()
  const wrapper = shallow(
    <WrappedComponent {...basicState} performSearch={myPerformSearch} />
  )

  expect(_isFunction(wrapper.props().fetchResults)).toBe(true)
})

test("mapDispatchToProps maps function performSearch", () => {
  const dispatch = jest.fn(() => Promise.resolve())
  const mappedProps = mapDispatchToProps(dispatch, {})

  mappedProps.performSearch("searchName", "query", "searchProjects")
  expect(dispatch).toBeCalled()
  expect(performSearch).toBeCalledWith("searchName", "query", "searchProjects")
  expect(mappedProps).toMatchSnapshot()
})
