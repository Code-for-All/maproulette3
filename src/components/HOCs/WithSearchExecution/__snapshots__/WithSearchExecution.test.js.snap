// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDispatchToProps maps function performSearch 1`] = `
Object {
  "performSearch": [Function],
}
`;

exports[`searchQueries.searchName is passed through to the wrapped component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Component
    searchQueries={
        Object {
            "searchName": "my search",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "0": "m",
      "1": "y",
      "2": " ",
      "3": "s",
      "4": "e",
      "5": "a",
      "6": "r",
      "7": "c",
      "8": "h",
      "fetchResults": [Function],
      "searchQueries": Object {
        "searchName": "my search",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "0": "m",
        "1": "y",
        "2": " ",
        "3": "s",
        "4": "e",
        "5": "a",
        "6": "r",
        "7": "c",
        "8": "h",
        "fetchResults": [Function],
        "searchQueries": Object {
          "searchName": "my search",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
